src/crystal/system/unix/process.cr

```diff
@@ -352,7 +352,7 @@ struct Crystal::System::Process
 
   private def self.reopen_io(src_io : IO::FileDescriptor, dst_io : IO::FileDescriptor)
     if src_io.closed?
-      Crystal::EventLoop.current.remove(dst_io)
+      typeof(Crystal::EventLoop.current).remove(dst_io)
```

Comment:
**issue:** Currently `EventLoop.current` is defined as a single type at compile time. But that could change in the future if we support two event loops with a runtime switch (e.g. `io_uring` and `epoll`).
Maybe it would be better to have a method `Crystal::EventLoop.remove(FileDescriptor)` to deal with this in a central place? It would also simplify the call sites.

Reply:
Right, it will be reduced as EventLoop+ and thus always call `EventLoop.resume` and never call `EventLoop::Polling.resume`. Damn.

Reply:
It's gonna be hard to avoid that if we want the even loop configurable at runtime 😢
But it should always be possible to select only a single event loop driver at compile time, which avoids multi dispatch, sacrificing flexibility for performance.

Reply:
I meant the evloop macro selector that's already repeated... but I introduced a `.backend_class` that should do the job.